### ISSUE TITLE
Stonesense doesn't receive input on OS X

### ISSUE THREAD

**Issue by lethosor** [DFHack/stonesense#32]
All input to the Stonesense window is passed to DF (regardless of whether the overlay is in use or not).
@danaris

**Comment by lethosor** [comment]
Neither of the causes I investigated (outdated allegro libraries and an issue with interposing SDL_PollEvent) seem to be the cause of this.

**Comment by Stefan-Zukin** [comment]
I'm having the same issue. No keyboard controls work, but it seems mouse controls work. Using LNP on OSX 10.11. Overall not too big an issue, still get the visual and all that works well, but it would be nice to be able to control it a bit.

**Comment by JapaMala** [comment]
Yeah, this isn't fixable without a complete re-write of stonesense, which
I've started, but I don't expect to have anything working anytime soon.

On 25-Jul-2016 12:28 pm, "Stefan Zukin" <notifications@github.com> wrote:

I'm having the same issue. No keyboard controls work, but it seems mouse
controls work. Using LNP on OSX 10.11. Overall not too big an issue, still
get the visual and all that works well, but it would be nice to be able to
control it a bit.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/DFHack/stonesense/issues/32#issuecomment-234856685>,
or mute
the thread

**Comment by Stefan-Zukin** [comment]
Okay. As is I got to see the chaos created by opening the floodgate on my dwarf-made waterfall and pump system, so I'm pretty happy with it. :)

**Comment by lethosor** [comment]
Well, it's mainly making Stonesense run standalone that's the bulk of the work (although it's not easy). @JapaMala, doesn't it already use RemoteFortressReader for some things?

**Comment by JapaMala** [comment]
Only very few things, like tree raws.

**Comment by Equilibrium94** [comment]
I'm having a similar problem on Windows. When I run stonesense with no overlay the controls work fine. However, when I run stone sense with overlay I can't figure out how I can do things like zoom in or zoom out or cut walls etc. Is it possible to pass input to stonesense with overlay on windows? Or even Linux?

**Comment by lethosor** [comment]
Because the overlay is running in the DF window, all input goes to DF (except maybe mouse input in the overlay area). Not sure there's a clean way to pass keyboard input to stonesense in any case - did you have something in mind?

**Comment by Equilibrium94** [comment]
Hmm I would really like to be able to zoom in. Maybe there is a config file I can set to determine default zoom level? It would also be nice to pull down the walls. The "c" key usually does that. Since, we can only get mouse input, would it be too hard to make double click replace the "c" command? Or perhaps you can think of a "dirty" way to make the controls work that wouldn't take too much effort? I don't mind difficult to use interface, but having to play zoomed out is kinda rough for me.

**Comment by lethosor** [comment]
I haven't actually worked with the overlay (not sure if anyone who has is still active); I just know how it works at a high level. It was released as an experimental feature and never really got finished.
For zooming, I'd think the scroll wheel would work pretty well, and it shouldn't be too hard to restrict it to when the mouse is over the overlay window itself (at least it's not hard with DF's SDL layer; Stonesense bypasses that and uses Allegro, which I've never really worked with).
